src/main/java/com/fzi/externalcontrol/impl/ExternalControlProgramNodeContribution.java

```diff
@@ -80,7 +80,11 @@ public boolean isDefined() {
   @Override
   public void generateScript(ScriptWriter writer) {
     String urScriptProgram = getInstallation().getUrScriptProgram();
+    String uniqueFunName = "fun_" + getInstallation().IncrementInstanceCounter() + "()";
+    writer.appendLine("def " + uniqueFunName + ":");
     writer.appendRaw(urScriptProgram);
+    writer.appendLine("end");
```

Comment:
When the program doesn't end with a newline, the `end` gets printed on the same line as the last line from the script due to the use of `appendRaw`. I think, that was actually the case since a long time, so strictly speaken this isn't a drawback of this PR. Just wanted to mention this so it doesn't get lost again.

```suggestion
    writer.appendRaw(urScriptProgram);
     writer.appendLine("");
    writer.appendLine("end");
```